### DISCUSSION TITLE
No provider for SASL mechanism GSSAPI: recompile librdkafka with libsasl2 or openssl support

### DISCUSSION THREAD

**Discussion by paritosh-96** [edenhill/librdkafka#3962] [Q&A]
I have a project connecting to Confluent Kafka and SARAMA kafka both. The connection fails for Confluent kafka only with the below error:
Failed to create Producer: No provider for SASL mechanism GSSAPI: recompile librdkafka with libsasl2 or openssl support. Current build options: PLAIN

I see this was faced by people earlier as well, and I have gone through those issues but could not resolve this.

I am trying to setup things in an Ubuntu system. Here is what my docker file looks like:

WORKDIR /opt
RUN apt-get update
ENV DEBIAN_FRONTEND=noninteractive
RUN apt-get -y install krb5-user libpam-krb5
ENV DEBIAN_FRONTEND=interactive
RUN apt-get -y install libsasl2-dev
RUN apt-get -y install libsasl2-modules-gssapi-mit
RUN \
    git clone https://github.com/edenhill/librdkafka.git && \
    cd librdkafka  && \
    ./configure --prefix /usr  && \
    make clean && \
    make && \
    make install && \
    export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:/usr/lib/pkgconfig/

After this I build my project with the -tags dynamic tag. I am using go 1.18.3. This seems to be working for other projects on 1.16 go version. 

Please help on this

**Comment by pranavrth** [comment]
Try using `./configure --enable-ssl --prefix` instead of just `./configure --prefix`

**Reply by paritosh-96** [reply]
@pranavrth   Thanks for the suggestion. I am still getting the error. The build was failing post the addition of this and asked to install libssl-dev as well. Post doing that the build worked but then I got the same error.
